semgrep/dependencyparser/find_lockfiles.py

```diff
@@ -13,16 +13,21 @@
 
 @functools.lru_cache(maxsize=None)
```

Comment:
just checked, and no, generators cannot be cached in the normal way lol. If you call this twice with the same arguments you get the pointer. I'll just remove it for now, since we only call it once anyway. If there ends up being a need for it, it's possible to do this using a custom decorator that uses `itertools.tee`.